tests/integration/targets/filter_to_prettytable/tasks/main.yml

```diff
@@ -0,0 +1,166 @@
+---
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Copyright (c) 2025, Timur Gadiev (tgadiev@gmail.com)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Install required libs
+  pip:
+    name: prettytable
+    state: present
+  delegate_to: localhost
+  become: false
+
+- name: Set test data
+  set_fact:
+    test_data:
+      - name: Alice
+        age: 25
+        role: admin
+      - name: Bob
+        age: 30
+        role: user
+    data_for_align:
+      - date: 2023-01-01
+        description: Office supplies
+        amount: 123.45
+
+# Test basic functionality
+- name: Test basic table creation
+  set_fact:
+    basic_table: '{{ test_data | community.general.to_prettytable }}'
+
+- name: Load expected basic table output
+  set_fact:
+    expected_basic_table_raw: "{{ lookup('file', 'basic_table.txt') }}"
+
+- name: Remove copyright header from expected output
+  set_fact:
+    expected_basic_table: "{{ expected_basic_table_raw.split('\n')[4:] | join('\n') }}"
+
+- name: Verify basic table output
+  assert:
+    that:
+      - basic_table | trim == expected_basic_table | trim
```

Comment:
Why are we testing the output piped into `trim`? That could potentially hide some inconsistency in the output of the filter.

Reply:
Agree. Removed `trim` filter from tests.